pixeltable/globals.py

```diff
@@ -20,11 +20,74 @@
 
 _logger = logging.getLogger('pixeltable')
 
-
 def init() -> None:
     """Initializes the Pixeltable environment."""
     _ = Catalog.get()
 
+def _get_or_drop_existing_path(
+    path_str: str,
+    caller_context: catalog.SchemaObjectType,
```

Comment:
instead: schema_obj_cls: Type[SchemaObject] or maybe even created_cls or something like that

and get rid of SchemaObjectType

Reply:
Done. FWIW, I introduced SchemaObjectType mainly for Snapshot as it is not a class derived from SchemaObject.
<!-- Reviewable comment -OElsCLp3dasugay1C-K:b8qkdc5 -->
<!-- Sent from Reviewable.io -->